test/mocks/upstream/cluster_info.cc

```diff
@@ -52,7 +52,9 @@ MockClusterInfo::MockClusterInfo()
       .WillByDefault(ReturnPointee(&max_requests_per_connection_));
   ON_CALL(*this, stats()).WillByDefault(ReturnRef(stats_));
   ON_CALL(*this, statsScope()).WillByDefault(ReturnRef(stats_store_));
-  ON_CALL(*this, transportSocketFactory()).WillByDefault(ReturnRef(*transport_socket_factory_));
+  ON_CALL(*this, transportSocketFactory())
```

Comment:
TLDR: Due to circular reference issues, this mock doesn't include an actual mock transport factory. This ON_CALL() is also bogus because it stores the reference at the time of construction, which is empty. The real fix is to just include the mock directly and fix the circular reference issues. I will add a TODO and fix this in a follow up.